modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java

```diff
@@ -75,7 +82,26 @@
      * @param ctx Kernal context.
      */
     public GridSystemViewManager(GridKernalContext ctx) {
-        super(ctx, ctx.config().getSystemViewExporterSpi());
+        super(ctx, ((Supplier<SystemViewExporterSpi[]>)() -> {
+            SystemViewExporterSpi[] spi = ctx.config().getSystemViewExporterSpi();
+
+            if (!IgniteComponentType.INDEXING.inClassPath())
+                return spi;
+
+            SystemViewExporterSpi[] spiWithSql = new SystemViewExporterSpi[spi != null ? spi.length + 1 : 1];
```

Comment:
Nevertheless, what if `spi` array already contains `SqlViewExporterSpi`?

Reply:
`SqlViewExporterSpi` now is package-private so it's forbidden for the user to create instances.

Reply:
Any way it is just somewhat sloppy.

Reply:
OK, I've added a check to ensure `SqlViewExporterSpi` not presented in the initial array.

Reply:
It's definitely redundant deu to a package privacy. I told about naming only. Please revert this check and merge.